pkg/client/unversioned/clientcmd/api/v1/types.go

```diff
@@ -54,7 +54,10 @@ type Preferences struct {
 type Cluster struct {
 	// Server is the address of the kubernetes cluster (https://hostname:port).
 	Server string `json:"server"`
-	// APIVersion is the preferred api version for communicating with the kubernetes cluster (v1, v2, etc).
+	// DEPRECATED: APIVersion is the preferred api version for communicating with the kubernetes cluster (v1, v2, etc).
+	//  Because a cluster can run multiple API groups and potentially multiple versions of each, it no longer makes sense to specify
+	//  a single value for the cluster version. This field isn't really needed anyway, so we're deprecating it without a replacement.
+	//  It will be ignored if it is present.
```

Comment:
> Will we need to express two preferred versions for one group?

I don't think we do and I'd prefer that we prevent that from happening.  Having a simple list of `PreferredVersion` doesn't prevent someone from expressing two preferred versions.  If we used a map, they couldn't do it.

Would you prefer the stronger structure or additional validation during save and use?